test/test_examples.py

```diff
@@ -223,3 +224,37 @@ def func(x: TensorType[sparse_coo]):
         func(rand(3, 4))
     with pytest.raises(TypeError):
         func(rand(3, 4).long())
+
+
+def test_example12():
+    @typechecked
+    def func(
+        feats: TensorType["batch":..., "annotator":3, "word", "feature"],
+        predicates: TensorType[
+            "batch":..., "annotator":3, "predicate":"word", "feature"
+        ],
+        pred_arg_pairs: TensorType[
+            "batch":..., "annotator":3, "predicate":"word", "argument":"word"
+        ],
+    ):
+        pass
+
+    func(ones(2, 1, 3, 4, 5), ones(2, 1, 3, 4, 5), ones(2, 1, 3, 4, 4))
+
+    # matches ...
+    with pytest.raises(TypeError):
+        func(ones(2, 1, 3, 4, 5), ones(2, 3, 4, 5), ones(2, 1, 3, 4, 4))
+    with pytest.raises(TypeError):
+        func(ones(2, 1, 3, 4, 5), ones(2, 2, 3, 4, 5), ones(2, 1, 3, 4, 4))
+
+    # annotator has 3
+    with pytest.raises(TypeError):
+        func(ones(2, 1, 2, 4, 5), ones(2, 1, 2, 4, 5), ones(2, 1, 2, 4, 4))
+
+    # predicate and argument match word
+    with pytest.raises(TypeError):
+        func(ones(2, 1, 3, 3, 5), ones(2, 2, 3, 4, 5), ones(2, 1, 3, 4, 4))
+    with pytest.raises(TypeError):
+        func(ones(2, 1, 3, 4, 5), ones(2, 2, 3, 3, 5), ones(2, 1, 3, 4, 4))
+    with pytest.raises(TypeError):
+        func(ones(2, 1, 3, 4, 5), ones(2, 2, 3, 4, 5), ones(2, 1, 3, 3, 4))
```

Comment:
The middle argument for these last three checks has size `2, 2, 3, 4, 5` when it should presumably have `2, 1, 3, 4, 5` to match the `...`.